WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,20 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    event.id.in?(self.events.map(&:id))
+  end
+
+  def belongs_to_region?(region_or_country)
```

Comment:
Hmm... the variable name `region_or_country` seems wrong here, isn't a country a kind of region? Shouldn't this just be `region` or possibly `country_or_continent`?